Add BOT class

teodorkg opened the pull request:
Implementation of binary ordered tree specialised in counting inversions in a vector of numbers

teodorkg committed:
Add BOT class

Implementation of binary ordered tree specialised in counting inversions in a vector of numbers

hhodzhov commented on BOT.cpp:
```diff
@@ -0,0 +1,152 @@
+template<class T>
+std::ostream& operator <<(std::ostream& os, const BOT<T>& bot)
+{
+	typename BOT<T>::Node* crr, *noSucc;
```
noSucc звучи малко странно..

hhodzhov commented on BOT.cpp:
```diff
@@ -0,0 +1,152 @@
+}
+
+template<class T>
+void BOT<T>::add(T el, Node*& crr)
```
не е ясно от пръв поглед какво точно се добавя... addElement би било по-добре + името на променливата "el"

hhodzhov commented on BOT.cpp:
```diff
@@ -0,0 +1,152 @@
+{
+	if (this != &other)
+	{
+		del(root);
```
може да промениш името на функцията del на deleteTree например, но ако след време няма да се объркаш, може да я оставиш така, понеже функцията е private

hhodzhov reviewed (commented):
Реализацията е добре, за наименованието смятам, че ще ги имаш предвид

teodorkg committed:
Add tests, fix bugs

Basic tests were added. There was a problem with the << operator on
empty tree input. Clarifications were given.
